airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/Destination.java

```diff
@@ -29,8 +35,129 @@ AirbyteMessageConsumer getConsumer(JsonNode config,
+                                                                        final Consumer<AirbyteMessage> outputRecordCollector)
+      throws Exception {
+    return new ShimToSerializedAirbyteMessageConsumer(getConsumer(config, catalog, outputRecordCollector));
+  }
+
   static void defaultOutputRecordCollector(final AirbyteMessage message) {
     System.out.println(Jsons.serialize(message));
   }
 
+  /**
+   * Backwards-compatibility wrapper for an AirbyteMessageConsumer. Strips the sizeInBytes argument away from the .accept call.
+   */
+  @Slf4j
+  class ShimToSerializedAirbyteMessageConsumer implements SerializedAirbyteMessageConsumer {
+
+    private final AirbyteMessageConsumer consumer;
+
+    public ShimToSerializedAirbyteMessageConsumer(final AirbyteMessageConsumer consumer) {
+      this.consumer = consumer;
+    }
+
+    @Override
+    public void start() throws Exception {
+      consumer.start();
+    }
+
+    /**
+     * Consumes an {@link AirbyteMessage} for processing.
+     * <p>
+     * If the provided JSON string is invalid AND represents a {@link AirbyteMessage.Type#STATE}
+     * message, processing is halted. Otherwise, the invalid message is logged and execution continues.
+     *
+     * @param inputString JSON representation of an {@link AirbyteMessage}.
+     * @throws Exception if an invalid state message is provided or the consumer is unable to accept the
+     *         provided message.
+     */
+    @Override
+    public void accept(final String inputString, final Integer sizeInBytes) throws Exception {
+      consumeMessage(consumer, inputString);
+    }
+
+    @Override
+    public void close() throws Exception {
+      consumer.close();
+    }
+
+    /**
+     * Consumes an {@link AirbyteMessage} for processing.
+     * <p>
+     * If the provided JSON string is invalid AND represents a {@link AirbyteMessage.Type#STATE}
+     * message, processing is halted. Otherwise, the invalid message is logged and execution continues.
+     *
+     * @param consumer An {@link AirbyteMessageConsumer} that can handle the provided message.
+     * @param inputString JSON representation of an {@link AirbyteMessage}.
+     * @throws Exception if an invalid state message is provided or the consumer is unable to accept the
+     *         provided message.
+     */
+    @VisibleForTesting
+    static void consumeMessage(final AirbyteMessageConsumer consumer, final String inputString) throws Exception {
+
+      final Optional<AirbyteMessage> messageOptional = Jsons.tryDeserialize(inputString, AirbyteMessage.class);
+      if (messageOptional.isPresent()) {
+        consumer.accept(messageOptional.get());
+      } else {
+        if (isStateMessage(inputString)) {
+          throw new IllegalStateException("Invalid state message: " + inputString);
+        } else {
+          log.error("Received invalid message: " + inputString);
+        }
```

Comment:
👍 I believe that all other types of messages that a source might emit (log, control, etc) should not be passed to the destination, so STATE messages are the only other types to check for which are meaningful